src/test/java/uk/gov/hmcts/reform/sscs/service/HearingsServiceTest.java

```diff
@@ -0,0 +1,187 @@
+            .build();
+
+        hearingsService = new HearingsService(hmcHearingApi, ccdCaseService, idamService);
+    }
+
+    @DisplayName("When wrapper with a valid Hearing State is given addHearingResponse should run without error")
+    @ParameterizedTest
+    @CsvSource(value = {
+        "CREATE_HEARING",
+        "UPDATE_HEARING",
+        "UPDATED_CASE",
+        "PARTY_NOTIFIED",
+    }, nullValues = {"null"})
+    void processHearingRequest(HearingState state) {
+        wrapper.setState(state);
+        assertThatNoException().isThrownBy(() -> hearingsService.processHearingRequest(wrapper));
+    }
+
+    @DisplayName("When wrapper with a valid cancel Hearing State is given addHearingResponse should run without error")
+    @Test
+    void processHearingRequest() {
+        given(idamService.getIdamTokens()).willReturn(IdamTokens.builder().idamOauth2Token(IDAM_OAUTH2_TOKEN).serviceAuthorization(SERVICE_AUTHORIZATION).build());
+
+        given(hmcHearingApi.deleteHearingRequest(any(), any(), any(), any())).willReturn(HearingResponse.builder().build());
+
+        wrapper.setState(CANCEL_HEARING);
+
+        assertThatNoException().isThrownBy(() -> hearingsService.processHearingRequest(wrapper));
+    }
+
+    @DisplayName("When wrapper with a invalid Hearing State is given addHearingResponse should throw an UnhandleableHearingState error")
+    @ParameterizedTest
+    @CsvSource(value = {
+        "null",
+    }, nullValues = {"null"})
+    void processHearingRequestInvalidState(HearingState state) {
+        wrapper.setState(state);
+
+        UnhandleableHearingStateException thrown = assertThrows(UnhandleableHearingStateException.class, () -> {
+            hearingsService.processHearingRequest(wrapper);
+        });
+
+        assertThat(thrown.getMessage()).isNotEmpty();
+    }
+
+
+    @DisplayName("When wrapper with a valid HearingResponse is given updateHearingResponse should return updated valid HearingResponse")
+    @ParameterizedTest
+    @CsvSource(value = {
+        "CREATE_HEARING,CREATE_HEARING",
+        "UPDATED_CASE,UPDATED_CASE",
+    }, nullValues = {"null"})
+    void updateHearingResponse(HearingState state, HearingEvent event) throws UpdateCaseException {
+        given(ccdCaseService.updateCaseData(any(SscsCaseData.class), any(EventType.class), anyString(), anyString())).willReturn(expectedCaseDetails);
+
+        wrapper.setState(state);
+
+        HearingResponse response = HearingResponse.builder().versionNumber(VERSION).hearingRequestId(HEARING_REQUEST_ID).build();
+        assertThatNoException().isThrownBy(() -> hearingsService.hearingResponseUpdate(wrapper, response));
+
+        verify(ccdCaseService, times(1)).updateCaseData(any(SscsCaseData.class), eq(event.getEventType()), eq(event.getSummary()), eq(event.getDescription()));
+
+        assertThat(wrapper.getCaseData().getEvents()).isNotEmpty();
+        assertThat(wrapper.getCaseData().getEvents().get(0).getValue().getType()).isNotEmpty();
```

Comment:
Optional: This would look a bit nicer if we use a variable for `wrapper.getCaseData().getEvents().get(0).getValue()`